integration/all/integration_test.go

```diff
@@ -0,0 +1,295 @@
+	topTimeoutInSeconds              = 10
+)
+
+var (
+	// supportedContainerRuntimes = []string{ContainerRuntimeDocker, ContainerRuntimeContainerd, ContainerRuntimeCRIO}
+	supportedK8sDistros = []string{K8sDistroAKSAzureLinux, K8sDistroAKSUbuntu, K8sDistroARO, K8sDistroMinikubeGH}
+	cleaningUp          = uint32(0)
+
+	// containerRuntime = flag.String("container-runtime", "", "allows to do validation for expected runtime in the tests")
+	image          = flag.String("image", "", "gadget container image")
+	integration    = flag.Bool("integration", false, "run integration tests")
+	dnsTesterImage = flag.String("dnstester-image", "ghcr.io/inspektor-gadget/dnstester:latest", "dnstester container image")
+	testComponent  = flag.String("test-component", "", "run tests for specific component")
+
+	doNotDeployIG  = flag.Bool("no-deploy-ig", false, "don't deploy Inspektor Gadget")
+	doNotDeploySPO = flag.Bool("no-deploy-spo", false, "don't deploy the Security Profiles Operator (SPO)")
+
+	k8sDistro = flag.String("k8s-distro", "", "allows to skip tests that are not supported on a given Kubernetes distribution")
+	k8sArch   = flag.String("k8s-arch", "amd64", "allows to skip tests that are not supported on a given CPU architecture")
+
+	gadgetRepository = flag.String("gadget-repository", "ghcr.io/inspektor-gadget/gadget", "repository where gadget images are stored")
+	gadgetTag        = flag.String("gadget-tag", "latest", "tag used for gadgets's OCI images")
+)
+
+var (
+	containerRuntime string
+	isDockerRuntime  bool
+)
+
+func cleanupFunc(cleanupCommands []*Command) {
+	if !atomic.CompareAndSwapUint32(&cleaningUp, 0, 1) {
+		return
+	}
+
+	fmt.Println("Cleaning up...")
+
+	// We don't want to wait for each cleanup command to finish before
+	// running the next one because in the case the workflow run is
+	// cancelled, we have few seconds (7.5s + 2.5s) before the runner kills
+	// the entire process tree. Therefore, let's try to, at least, launch
+	// the cleanup process in the cluster:
+	// https://docs.github.com/en/actions/managing-workflow-runs/canceling-a-workflow#steps-github-takes-to-cancel-a-workflow-run
+	for _, cmd := range cleanupCommands {
+		err := cmd.StartWithoutTest()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%s\n", err)
+		}
+	}
+
+	for _, cmd := range cleanupCommands {
+		err := cmd.WaitWithoutTest()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%s\n", err)
+		}
+	}
+}
+
+func testMainInspektorGadget(m *testing.M) int {
+	if os.Getenv("KUBECTL_GADGET") == "" {
+		fmt.Fprintf(os.Stderr, "please set $KUBECTL_GADGET.")
+		return -1
+	}
+
+	if *k8sDistro != "" {
+		found := false
+		for _, val := range supportedK8sDistros {
+			if *k8sDistro == val {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			fmt.Fprintf(os.Stderr, "Error: invalid argument '-k8s-distro': %q. Valid values: %s\n",
+				*k8sDistro, strings.Join(supportedK8sDistros, ", "))
+			return -1
+		}
+	}
```

Comment:
This should be done also for `IgTestComponent`, so I suggest moving it to `initialize` where you're parsing and validating flags in common.